### PR TITLE
State manager

### LINKED ISSUE #9: State manadger
integrate state manager Jotai;

### PR DESCRIPTION
**Pull request by KirillSerg**
fix #9
re-write handling events with Jotai

### CODE REVIEW

File: src/components/SingleElement.tsx [hunk 1]
```diff
@@ -1,75 +1,43 @@
...
+    <>
+      {element.type !== 'free' && (
+        <element.type //flexible&dynemic rendering svg-elements
+          {...element}
+          style={{ cursor: 'pointer' }}
+          onMouseDown={(e) => handleMouseDown(e)}
+          onMouseUp={onMouseUp}
+        />
+      )}
+    </>
```
Comment (roman01la) [inline src/components/SingleElement.tsx:40]: the fragment is not needed here

File: src/store/store.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      const activeElementType = get(activeElementTypeAtom)
+      const newEl = { ...initialElement, type: activeElementType, id: crypto.randomUUID() }
+      set(elementsAtom, (prev) => [...prev, newEl])
+      set(selectedElementAtom, newEl)
+    }
+    set(activeElementTypeAtom, "free")
+    set(isDraggingAtom, false)
+    set(selectingAreaAtom, null)
+  }
+)
```
Comment (roman01la) [inline src/store/store.ts:98]: looks good, although I'd suggest to use a single state object instead of many small ones, but it's up to you

**Review by roman01la — APPROVED** [review]

File: src/components/SingleElement.tsx (hunk 1 above)
Comment (KirillSerg) [reply src/components/SingleElement.tsx:40]: image
image

File: src/store/store.ts (hunk 2 above)
Comment (KirillSerg) [reply src/store/store.ts:98]: the documentation advises using smaller atoms. So it is still easier to navigate. If you don't mind, I'll leave it for now

**Pull request merged**